x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/RoleDescriptorsIntersection.java

```diff
@@ -70,7 +70,7 @@ public static RoleDescriptorsIntersection fromXContent(XContentParser xContentPa
             while ((token = p.nextToken()) != XContentParser.Token.END_OBJECT) {
                 XContentParserUtils.ensureExpectedToken(XContentParser.Token.FIELD_NAME, token, p);
                 XContentParserUtils.ensureExpectedToken(XContentParser.Token.START_OBJECT, p.nextToken(), p);
-                roleDescriptors.add(RoleDescriptor.parse(p.currentName(), p, false));
+                roleDescriptors.add(RoleDescriptor.parser().allowRestriction(true).parse(p.currentName(), p));
```

Comment:
Might extract this into a private static field to avoid instantiating every time we parse (here in other prod code spots).